wgpu-core/src/device/mod.rs

```diff
@@ -998,15 +998,31 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let (format, view_kind, range) = match desc {
             Some(desc) => {
                 let kind = conv::map_texture_view_dimension(desc.dimension);
+                let level_count = desc.base_mip_level + desc.level_count;
```

Comment:
If the `level_count` is zero here, and `base_mip_level` == maximum - 1, we don't want the validation to pass.
We need to have something like `desc.level_count.max(1)` here